src/network/wait-online/manager.c

```diff
@@ -41,6 +41,9 @@ bool manager_ignore_link(Manager *m, Link *link) {
         if (m->interfaces && !strv_contains(m->interfaces, link->ifname))
                 return true;
 
+        if (link->required_for_online == 0)
```

Comment:
required_for_online is a boolean, so please use `!link->required_for_online` instead.